presto-common/src/test/java/com/facebook/presto/common/block/TestBlockBuilderUtil.java

```diff
@@ -0,0 +1,145 @@
+        writePositionToBlockBuilder(expectedBlock, 0, blockBuilder2);
+        Block newBlock = blockBuilder2.build();
+        assertEquals(newBlock, expectedBlock);
+    }
+
+    @Test
+    public void testMapBlockBuilder()
+    {
+        BlockBuilder blockBuilder1 = TEST_MAP_TYPE.createBlockBuilder(null, 1);
+        BlockBuilder mapBlockBuilder = blockBuilder1.beginBlockEntry();
+        writeValuesToMapBuilder(mapBlockBuilder);
+        Block expectedBlock = blockBuilder1.closeEntry().build();
+
+        // write values to a new block using BlockBuilderUtil
+        BlockBuilder blockBuilder2 = TEST_MAP_TYPE.createBlockBuilder(null, 1);
+        writePositionToBlockBuilder(expectedBlock, 0, blockBuilder2);
+        Block newBlock = blockBuilder2.build();
+        assertEquals(newBlock, expectedBlock);
+    }
+
+    @Test
+    public void testRowBlockBuilder()
+    {
+        RowType rowType = rowType(VARCHAR, BIGINT, TEST_MAP_TYPE);
+        BlockBuilder blockBuilder = rowType.createBlockBuilder(null, 1);
+
+        BlockBuilder rowBlockBuilder = blockBuilder.beginBlockEntry();
+        VARCHAR.writeString(rowBlockBuilder, "TEST_ROW");
+        BIGINT.writeLong(rowBlockBuilder, 10L);
+        BlockBuilder mapBlockBuilder = rowBlockBuilder.beginBlockEntry();
+        writeValuesToMapBuilder(mapBlockBuilder);
+        rowBlockBuilder.closeEntry();
+        Block expectedBlock = blockBuilder.closeEntry().build();
+
+        // write values to a new block using BlockBuilderUtil
+        BlockBuilder blockBuilder2 = rowType.createBlockBuilder(null, 1);
+        writePositionToBlockBuilder(expectedBlock, 0, blockBuilder2);
+        Block newBlock = blockBuilder2.build();
+        assertEquals(newBlock, expectedBlock);
+    }
+
+    private static void writeValuesToMapBuilder(BlockBuilder mapBlockBuilder)
+    {
+        for (Map.Entry<Long, String> entry : TEST_MAP_VALUES.entrySet()) {
+            Long key = entry.getKey();
+            if (key == null) {
+                mapBlockBuilder.appendNull();
+            }
+            else {
+                BIGINT.writeLong(mapBlockBuilder, entry.getKey());
+            }
+
+            String value = entry.getValue();
+            if (value == null) {
+                mapBlockBuilder.appendNull();
+            }
+            else {
+                VARCHAR.writeString(mapBlockBuilder, entry.getValue());
+            }
+        }
+    }
+
+    public static RowType rowType(Type... fieldTypes)
+    {
+        List<RowType.Field> fields = new ArrayList<>();
+        for (int i = 0; i < fieldTypes.length; i++) {
+            Type fieldType = fieldTypes[i];
+            String fieldName = "field_" + i;
+            fields.add(new RowType.Field(Optional.of(fieldName), fieldType));
+        }
+        return RowType.from(fields);
+    }
+
+    public static void throwUnsupportedOperation()
+    {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Do you need this ?

Reply:
It is used via reflection for creating mapType. Added a comment here for clarity.